static/src/javascripts/projects/commercial/modules/hosted/youtube.js

```diff
@@ -0,0 +1,102 @@
+// @flow
+import nextVideoAutoplay from 'commercial/modules/hosted/next-video-autoplay';
+import { initYoutubePlayer } from 'common/modules/atoms/youtube-player';
+import tracking from 'common/modules/atoms/youtube-tracking';
+import detect from 'lib/detect';
+import mediator from 'lib/mediator';
+
+type videoPlayerComponent = { getCurrentTime: () => number };
+
+const EVENTSFIRED = [];
+
+const isDesktop = (): boolean => detect.isBreakpoint({ min: 'desktop' });
+
+const sendPercentageCompleteEvents = (
+    atomId: string,
+    youtubePlayer: videoPlayerComponent,
+    playerTotalTime: number
+): void => {
+    const quartile = playerTotalTime / 4;
+    const playbackEvents = {
+        '25': quartile,
+        '50': quartile * 2,
+        '75': quartile * 3,
+    };
+
+    Object.keys(playbackEvents).forEach(key => {
+        const value = playbackEvents[key];
```

Comment:
Niiiice

```
    Object.entries(playbackEvents).forEach(([key, value]) => {
        if (
            !EVENTSFIRED.includes(key) &&
            youtubePlayer.getCurrentTime() > Number(value)
        ) {
            tracking.track(key, atomId);
            EVENTSFIRED.push(key);
            mediator.emit(key);
        }
    });
```
_________

Just tried this out in Safari; it throws: `TypeError: Object.entries is not a function.` 
... so looks like we don't yet have a polyfill for `Object.entries`. 
😞 😞